### PR TITLE
splitting out pkg/sync and pkg/startup for multi-version plugin

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
Work towards ability to update clusters which are on a supported old version
with respect to the RP/plugin.

* Set master branch to work towards v4.0; we will reset this to v5.0 when we
  branch release-v4

* Make pluginconfig support multiple versions in parallel - see
  /pluginconfig/pluginconfig-311.yaml as an example

* Rename pkg/addons -> pkg/sync, simplify to an interface and make multiversion

* Split pkg/arm -> pkg/startup, simplify to an interface and make multiversion

* Move lint-addons to a unit test

* Reduce dependencies on pkg/cluster in preparation for making it multiversion

* Miscellaneous sync pod performance, behaviour and code style fixes
```

### CODE REVIEW

**Comment by jim-minter** [comment]
@mjudeikis here's the next instalment, ptal - the main piece left now is to make pkg/{config,arm} multiversion

File: pkg/fakerp/config.go [hunk 1]
```diff
@@ -61,43 +61,47 @@ func GetPluginTemplate() (*pluginapi.Config, error) {
 	template.Certificates.GenevaLogging.Key = logKey
 	template.Certificates.GenevaMetrics.Cert = metCert
 	template.Certificates.GenevaMetrics.Key = metKey
-	template.Images.GenevaImagePullSecret = pullSecret
-	template.Images.ImagePullSecret = imagePullSecret
+	template.GenevaImagePullSecret = pullSecret
+	template.ImagePullSecret = imagePullSecret
```
Comment (mjudeikis) [inline pkg/fakerp/config.go]: This is sole image config. This is why it was under `Images`. Any reasoning to move/change api?

**Comment by mjudeikis** [comment]
looks fine. 
/retest

File: pkg/fakerp/config.go (hunk 1 above)
Comment (jim-minter) [reply pkg/fakerp/config.go]: yes - I am hoping the pull secrets are the same regardless of the cluster version

**Comment by mjudeikis** [comment]
want to merge it as it is?

**Comment by jim-minter** [comment]
/hold
scale up/down still rotating

**Comment by jim-minter** [comment]
green across the board, nice!
adding lgtm as @mjudeikis said "looks fine" before :)

**Pull request merged**